### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by NilsHullegien**
This branch adds the tests needed to get our project to > 75% coverage.

### CODE REVIEW

**Comment by LMiljak** [comment]
Please, Please do smaller PR's next time.
2800 line additions and 600 deletions is A LOT.

File: ContextDesktop/invalid [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<scores>
+</scores>
+<scores>
+</scores>
```
Comment (LMiljak) [inline ContextDesktop/invalid]: What is this?

**Comment by NilsHullegien** [comment]
I am very sorry about that, you are absolutely right. Next time I will create a PR every couple of classes tested/every package.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/main/LobbyHUDController.java [hunk 2]
```diff
@@ -111,22 +111,30 @@ private void addInstruction() {
...
     }
 
+    /**
+     * Getter for the players.
+     * Used in testing ONLY.
+     * @return the players hashmap.
+     */
+    public HashMap<PlatformPosition, BitmapText> getPlayers() {
+        return players;
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/main/LobbyHUDController.java:139]: You can make this method protected.
While it still doesn't completely ensure that other classes in the same package (other than test classes) won't touch it, at least classes in other packages definately won't use it.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/main/Main.java [hunk 3]
```diff
@@ -313,7 +311,15 @@ public boolean isBugEventRunning() {
...
+    }
 
+    /**
+     * Sets the serverwrapper.
+     * Used in testing ONLY.
+     * @param newServerWrapper the new serverWrapper
+     */
+    public void setServerWrapper(ServerWrapper newServerWrapper) {
+        server = newServerWrapper;
     }
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/main/Main.java:323]: Again, you could make this protected.

File: ContextDesktop/invalid (hunk 1 above)
Comment (NilsHullegien) [inline ContextDesktop/invalid]: Whoops, that was a file that was generated in an earlier test. Will remove before merge.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/CarrierAssigner.java [hunk 4]
```diff
@@ -51,10 +51,10 @@ public void connectionAdded(Server server, HostedConnection conn) {
...
 	public void connectionRemoved(Server server, HostedConnection conn) {
-		Logger.getGlobal().log(Level.INFO, conn.getAddress() + " has disconnected");
-		for (PlatformPosition position : PlatformPosition.values()) {
-			carrierAddressMap.remove(position, conn);
-		}
+        Logger.getGlobal().log(Level.INFO, conn.getAddress() + " has disconnected");
+        for (PlatformPosition position : PlatformPosition.values()) {
+        	carrierAddressMap.remove(position, conn);
+        }
 	}
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/CarrierAssigner.java]: I'm not sure the formatting is right in this method.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/server/ClientFinder.java [hunk 5]
```diff
@@ -115,7 +117,7 @@ private void findClients() {
 			}
 
 		} catch (IOException e) {
-		    System.out.println(e.getStackTrace());
+		    e.getStackTrace();
 		}
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/server/ClientFinder.java:121]: This line of code does absolutely nothing. Can't you use e.printStackTrace()?

File: ContextDesktop/src/test/java/com/github/migi_1/Context/model/entity/behaviour/TestMovingObstacleMoveBehaviour.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    
+    /**
+     * Tests if the moveVector gets updated when updateMoveVector method gets called.
+     */
+    @Test
+    public void testUpdateMoveVectors() {
+        moveBehaviour.updateMoveVector();
+        Mockito.verify(model, Mockito.times(2)).getLocalTranslation();
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/model/entity/behaviour/TestMovingObstacleMoveBehaviour.java:56]: I'm not sure why getLocalTranslation() is getting verified in this test.
This test method's javadoc says "Tests if the moveVector gets updated when updateMoveVector method gets called.". 
The getLocalTranslation() method doesn't give any indication that it got updated.

File: ContextMessages/src/test/java/com/github/migi_1/ContextMessages/TestImmobilisedMessage.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	/**
+	 * Tests the empty constructor.
+	 */
+	@Test
+	public void testConstructor() {
+	    assertEquals(ImmobilisedMessage.class, new ImmobilisedMessage().getClass());
+	    assertEquals(ImmobilisedMessage.class, message.getClass());
+	}
+
+}
```
Comment (LMiljak) [inline ContextMessages/src/test/java/com/github/migi_1/ContextMessages/TestImmobilisedMessage.java:49]: Formatting feels weird in this class at some places.

File: ContextMessages/src/test/java/com/github/migi_1/ContextMessages/TestPositionMessage.java [hunk 8]
```diff
@@ -11,21 +11,30 @@
...
+
+	/**
+	 * Tests the constructors.
+	 */
+	@Test
+    public void constructorTest() {
+        assertEquals(PositionMessage.class, new PositionMessage().getClass());
+        assertEquals(PositionMessage.class, new PositionMessage(PlatformPosition.FRONTRIGHT).getClass());
+    }
+
```
Comment (LMiljak) [inline ContextMessages/src/test/java/com/github/migi_1/ContextMessages/TestPositionMessage.java:40]: Formatting feels weird in this method.

**Comment by NilsHullegien** [comment]
I will definitely fix these issues, closing PR for now.

**Comment by NilsHullegien** [comment]
Formatting is fixed now, some methods are protected now.
Reopening PR.

File: ContextDesktop/src/test/java/com/github/migi_1/Context/model/entity/behaviour/TestMovingObstacleMoveBehaviour.java (hunk 6 above)
Comment (NilsHullegien) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/model/entity/behaviour/TestMovingObstacleMoveBehaviour.java:56]: You are definitely right, I will change it so it checks that the vector is changed.

**Comment by LMiljak** [comment]
I approve now :+1:

**Comment by dVoorhout** [comment]
Because of the size of this PR it's very hard to look at everything in detail, but I didn't spot any significant or obvious mistakes

**Pull request merged**